src/transformers/models/univnet/modeling_univnet.py

```diff
@@ -0,0 +1,654 @@
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_start_docstrings,
+    logging,
+)
+from .configuration_univnet import UnivNetGanConfig
+
+
+logger = logging.get_logger(__name__)
+
+
+_CHECKPOINT_FOR_DOC = "dg845/univnet-dev"
+
+UNIVNET_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "dg845/univnet-dev",
+    # See all UnivNet models at https://huggingface.co/models?filter=univnet
+]
+
+
+class UnivNetKernelPredictorResidualBlock(nn.Module):
+    """
+    Implementation of the residual block for the kernel predictor network inside each location variable convolution
+    block (LVCBlock).
+
+    Parameters:
+        channels ('int', *optional*, defaults to 64):
+            The number of hidden channels for the residual block.
+        kernel_size (`int`, *optional*, defaults to 3):
+            The kernel size for the 1D convolutional layers in the residual block.
+        dropout (`float`, *optional*, defaults to 0.0):
+            The dropout probability for the Dropout layer in the residual block.
+        leaky_relu_slope (`float`, *optional*, defaults to 0.1):
+            The angle of the negative slope used by the leaky ReLU activation.
+    """
+
+    def __init__(
+        self,
+        channels: int = 64,
+        kernel_size: int = 3,
+        dropout: float = 0.0,
+        leaky_relu_slope: float = 0.1,
```

Comment:
I guess I'm not sure about this because I think annotating the type as `Optional[...]` implies that `None` itself is a valid input for that argument (see [this SO thread](https://stackoverflow.com/questions/51710037/how-should-i-use-the-optional-type-hint)). For example, in the above code snippet, `channels` is passed to [`torch.nn.Conv1d`](https://pytorch.org/docs/stable/generated/torch.nn.Conv1d.html) as both `in_channels` and `out_channels`, which I don't think accept `None` as an input. So I think annotating the type with `Optional[...]` should happen only if the default argument is `None`, or passing in `None` is explicitly handled inside the function.

AFAIK, using a "bare" type annotation like `int` when supplying a default value and then describing the argument as optional in the dosctring is common in `diffusers` (see e.g. [here](https://github.com/huggingface/diffusers/blob/a7de96505b8c71a626c6fa45ff3cf85e42db4960/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L529)). Is the practice in `transformers` different?

Reply:
In my understanding, using `Optional[...]` implies to the user that he don't really need to pass the parameter value, which is the case here since you defined default values. This is how it is done in transformers!

Reply:
No, that is not how we use it in Transformers. `Optional[Xxx]` as a type annotation means `None` is a possible value, not that a default value is provided (for that we say optional in the docstring).

Reply:
Thanks @sgugger for clarifying this point.